Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -633,12 +636,28 @@ private void EnsureColumnBinding(object dataItem)
                 throw DataGridError.DataGridComboBoxColumn.UnsetBinding(GetType());
             }
 
-            if (!dataItem?.GetType().GetProperties().Any(x => x.Name.Equals(Binding.Path.Path)) ?? false)
+            var property = dataItem.GetType().GetNestedProperty(Binding.Path.Path);
+
+            if (property == null)
             {
                 throw DataGridError.DataGridComboBoxColumn.UnknownBindingPath(Binding, dataItem.GetType());
             }
         }
 
+        private void EnsureColumnTypeAgreement(object dataItem)
+        {
+            if (string.IsNullOrEmpty(DisplayMemberPath))
+            {
+                var itemsSourceType = ItemsSource.GetType().GetGenericArguments().First();
```

Comment:
What if ItemsSource is not a generic type? Shouldn't you use TypeHelper.GetEnumerableItemType instead?